src/validation.cpp

```diff
@@ -2704,6 +2704,9 @@ bool CChainState::ActivateBestChain(CValidationState &state, const CChainParams&
             // Block until the validation queue drains. This should largely
             // never happen in normal operation, however may happen during
             // reindex, causing memory blowup if we run too far ahead.
+            // Note that if a validationinterface callback ends up calling
+            // ActivateBestChain this may lead to a deadlock! We should
```

Comment:
Maybe add that this could lead to a deadlock "because ActiveBestChain won't be able to acquire m_cs_chainstate", if this is correct, to avoid making a puzzle of where the deadlock is.